client/src/App.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import './App.css';
+import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
+import Button from "@material-ui/core/Button";
+import theme from "./theme";
+import { makeStyles } from '@material-ui/core/styles';
+import Eventcard from './eventcard.js';
+
+
+// function App() {
+//   return (
+//     <div className="App">
+//       <header className="App-body">
+//
+//
+//         <Eventcard />
+//
+//       </header>
+//     </div>
+//   );
+// }
+
+
+class App extends Component {
+state = {
+    data: null
+  };
+
+  componentDidMount() {
+      // Call our fetch function below once the component mounts
+    this.callBackendAPI()
```

Comment:
While this will probably never be an issue for this component since it is App (and therefore won't be unmounted), there is a potential error here when setting state after the promise resolves.

Imagine this component is not `<App />` but `<EventCard />`. If we render an event card, it will reach out to the API as soon as it's mounted. However, if we decide to un-render the `<EventCard />` because we changed filters or navigated away, the component unmounts.

Unfortunately, the promise is still active. If it resolves, it will call setState, which will initiate a re-render on an unmounted component. This will result in an error, and may even give you a blank screen.

**Solution**
There are multiple ways to solve this. One of them is to handle this asynchronous call outside of the component, and pass the result in through props. The props won't affect the component unless it's mounted, so problem solved.

The other is to keep track of an `isMounted` variable, setting it to `true` in `componentDidMount()` and `false` in `componentDidUnmount()`. Then you can check the value before calling `setState`.

Reply:
I would like to go over this.